### PR TITLE
Rfairooz/topic

### LINKED ISSUE #81: Create Topic Component
image

- Create a topic component that will get a topic from props and put it into the container

### PR DESCRIPTION
**Pull request by rfairooz**
Screen Shot 2022-08-22 at 2 36 08 AM
Screen Shot 2022-08-22 at 2 35 21 AM
Screen Shot 2022-08-22 at 2 35 45 AM

### CODE REVIEW

File: components/Topic.js [hunk 1]
```diff
@@ -1,7 +1,22 @@
 import React from "react";
 
-const Topic = () => {
-	return <div>Topic</div>;
+const Topic = ({ Topic, BackgroundColor, TextColor }) => {
```
Comment (shahdivyank) [inline components/Topic.js]: camelCase for variables ie. `topic, backgroundColor, textColor`

File: components/Topic.js [hunk 2]
```diff
@@ -1,7 +1,22 @@
...
+	return (
+		<div className='flex w-full h-full'>
+			<div
+				className='self-center box-border h-24 w-48 border-2 border-black rounded-3xl py-3'
```
Comment (shahdivyank) [inline components/Topic.js]: don't hard card the height and width, once you merge the PR for the setup, you can go from there

File: components/Topic.js [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+				style={{
+					backgroundColor: BackgroundColor,
```
Comment (shahdivyank) [inline components/Topic.js]: use Tailwind CSS to apply these properties

File: components/Topic.js [hunk 4]
```diff
@@ -1,7 +1,22 @@
...
+					color: TextColor,
+					boxShadow: "8px -6px black",
+				}}
+			>
+				<div className='flex justify-center items-center py-3 font-lexend font-bold text-lg'>
+					{Topic}{" "}
```
Comment (shahdivyank) [inline components/Topic.js]: what is the `{" "}` for?

**Review by shahdivyank — CHANGES_REQUESTED** [review]

File: components/Topic.js (hunk 4 above)
Comment (rfairooz) [reply components/Topic.js]: oh when I run prettier it formats like this. Not sure why the "" is added.

Comment (shahdivyank) [reply components/Topic.js]: hmmm, look up the reason for that and put a rule in the prettier config file to fix that

File: components/Topic.js [hunk 5]
```diff
@@ -1,7 +1,27 @@
 import React from "react";
 
-const Topic = () => {
-	return <div>Topic</div>;
+const Topic = ({ topic, backgroundcolor, textcolor }) => {
```
Comment (shahdivyank) [inline components/Topic.js]: `backgroundColor` `textColor`

File: components/Topic.js [hunk 6]
```diff
@@ -1,7 +1,27 @@
...
+	return (
+		<div className='flex w-full h-full'>
+			<div
+				style={{
+					justifySelf: "center",
```
Comment (shahdivyank) [inline components/Topic.js]: use Tailwind CSS

**Review by shahdivyank — CHANGES_REQUESTED** [review]

File: components/Topic.js [hunk 7]
```diff
@@ -1,7 +1,39 @@
...
+					paddingBottom: "0.75rem",
+					// paddingRight: "0.25rem",
+					// paddingLeft: "0.25rem",
+					backgroundColor: backgroundColor,
+					color: textColor,
+					boxShadow: "8px -6px black",
+				}}
+			>
+				<div
+					style={{
```
Comment (shahdivyank) [inline components/Topic.js]: Convert to Tailwind CSS

**Review by shahdivyank — CHANGES_REQUESTED** [review]

File: pages/events.js [hunk 8]
```diff
@@ -1,12 +1,18 @@
...
 import Calendar from "../components/Calendar";
 import PastEvents from "../components/PastEvents";
+import Topic from "../components/Topic";
 
 const Events = () => {
 	return (
 		<div>
 			<Calendar />
 			<PastEvents />
+			<Topic
```
Comment (shahdivyank) [inline pages/events.js]: You can remove this, since we have the implementation under the `Learn` section

**Comment by shahdivyank** [comment]
Make sure to solve the merge conflicts

File: components/Topic.js (hunk 7 above)
Comment (rfairooz) [reply components/Topic.js]: wait I thought I did change everything to tailwind css. Could you explain what you mean here?

File: components/Learn.js [hunk 9]
```diff
@@ -4,23 +4,84 @@ import Topic from "../components/Topic";
...
+				<Col className='flex justify-center items-center'>
+					<Topic
+						topic='research'
+						backgroundColor='#E37A45'
+						textColor='black'
+					/>
+				</Col>
+			</Row>
+			<Row>
+				<Col></Col>
```
Comment (shahdivyank) [inline components/Learn.js]: There is an empty `Col` here

File: components/Topic.js [hunk 10]
```diff
@@ -1,7 +1,21 @@
...
-
-const Topic = () => {
-	return <div></div>;
+const Topic = ({ topic, backgroundColor, textColor }) => {
+	return (
+		<div>
+			<div
+				className='flex justify-center items-center w-96 h-36 rounded-3xl border-2 border-black mb-8 shadow-black'
+				style={{
+					backgroundColor: backgroundColor,
```
Comment (shahdivyank) [inline components/Topic.js]: use Tailwind here, you can do `bg-[backgroundColor]` the same can be done for text `text-[textColor]`

File: components/Topic.js [hunk 11]
```diff
@@ -1,7 +1,21 @@
 import React from "react";
-
-const Topic = () => {
-	return <div></div>;
+const Topic = ({ topic, backgroundColor, textColor }) => {
+	return (
+		<div>
+			<div
+				className='flex justify-center items-center w-96 h-36 rounded-3xl border-2 border-black mb-8 shadow-black'
```
Comment (shahdivyank) [inline components/Topic.js]: Height and width shouldn't be hard coded with w-96 and h-36, it should just take up space of the column it is in

File: components/Learn.js [hunk 12]
```diff
@@ -4,23 +4,84 @@ import Topic from "../components/Topic";
...
 const Learn = () => {
 	return (
-		<div className='font-lexend text-lg'>
-			{" "}
-			<div className='flex justify-center items-center font-lexend font-bold text-4xl mb-4'>
+		<div className='w-full font-lexend text-4xl'>
+			<div className='flex justify-center items-center font-lexend font-bold text-6xl mb-2'>
 				Learn
 			</div>
+			<div className='py-10'>
```
Comment (shahdivyank) [inline components/Learn.js]: You can move the div and just apply the padding on the Row, it is a large value so make sure to test it out on different screen sizes

**Review by shahdivyank — CHANGES_REQUESTED** [review]

**Review by shahdivyank — APPROVED** [review]

**Review by vwong031 — APPROVED** [review]

**Pull request merged**